### PR TITLE
Add new server-side same-origin option

### PR DESCRIPTION
**Pull request by tylerkelly13**
Related Ticket: DOC-467

Description of Changes:
* Document new same-origin option for server-side components
* Add 5.3 release note 
* Consolidate the server-side component release notes

DOD:
- [x] nav.yml has been updated if applicable
- [x] file has been included where required if applicable
- [x] files removed have been deleted, not just excluded from the build if applicable

Review
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -48,6 +48,8 @@ requires_5_1v: "> **Note**: This feature is only available for TinyMCE 5.1 and l
 requires_5_2v: "> **Note**: This feature is only available for TinyMCE 5.2 and later."
 requires_5_3v: "> **Note**: This feature is only available for TinyMCE 5.3 and later."
 
+requires_jsscwar_230v: "> **Note**: This feature is only available for TinyMCE self-hosted server-side components version 2.3.0 and later."
```
Comment (ephox-mogran) [inline _config.yml:51]: I guess we don't really care about hex as it's just an HCL bundle, but it's 2.4.0 for hex. Do you feel this wording is making it obvious that it's talking about each individual service war? e.g. `ephox-spelling`. Should we put it in a note of how to check your version?

File: _data/nav.yml [hunk 2]
```diff
@@ -727,9 +727,9 @@
     - url: "#Overview"
     - url: "#TinyMCE 5.3 new features and enhancements"
     - url: "#Accompanying Premium Plugin changes"
+    - url: "#Accompanying premium self-hosted server-side component changes"
```
Comment (ephox-mogran) [inline _data/nav.yml:730]: Why is `Premium` capitalised above but not here?

File: enterprise/server/configure.md [hunk 3]
```diff
@@ -101,12 +101,30 @@ The `*` wildcard character matches any value. Wildcards are supported in the fol
...
 }
 ````
 
+#### `allowed-origins.same-origin` (optional)
+
+{{site.requires_jsscwar_230v}}
+
+The `allowed-origins` `same-origin` option can be used to block all cross-origin requests. This option is set to `false` by default.
+
+Setting the `same-origin` setting to `true` will block all HTTP `OPTIONS` requests and allow all other HTTP methods. When set to `true`, all `origins` specified in `allowed-origins` will be ignored.
```
Comment (ephox-mogran) [inline enterprise/server/configure.md:116]: Do we need to include the purpose of why someone would want this setting?

File: release-notes/release-notes53.md [hunk 4]
```diff
@@ -63,42 +63,15 @@ For a list of valid menu items, see: [Editor control identifiers - Menu controls
 
 ### Enhanced Media Embed
```
Comment (ephox-mogran) [inline release-notes/release-notes53.md:65]: Where did all of these notes go? Somewhere else?

File: release-notes/release-notes53.md [hunk 5]
```diff
@@ -124,16 +97,49 @@ The {{site.productname}} 5.3 release includes an accompanying release of the **S
...
-- Deploying the server-side component, see: [Server-side component installation]({{site.baseurl}}/enterprise/server/).
-- Docker deployments of the server-side components, see: [Containerized service deployments]({{site.baseurl}}/enterprise/server/dockerservices/).
+- The Link Checker plugin, see: [Link Checker plugin]({{site.baseurl}}/plugins/linkchecker/).
+- The Image Tools plugin, see: [Image Tools plugin]({{site.baseurl}}/plugins/imagetools/).
+- The Enhanced Media Embed plugin, see: [Enhanced Media Embed plugin]({{site.baseurl}}/plugins/mediaembed/).
+- Deploying the server-side components, see: [Server-side component installation]({{site.baseurl}}/enterprise/server/).
+
+### Removal of Java 7 support
+
+Java 7 support has been removed with this release. The {{site.productname}} Java server-side components now require a minimum of Java 8. From the {{site.productname}} 5.3 release, the `.war` files provided in the self-hosted bundles will not be compatible with Java 7. This change resolves security issues present in the older versions of these services. The following plugins are affected by this change:
```
Comment (ephox-mogran) [inline release-notes/release-notes53.md:125]: More of a curiosity: should we mention the actual plugin name that integrators put into their config as well? Or is the full descriptive name sufficient?

File: release-notes/release-notes53.md (hunk 4 above)
Comment (metricjs) [reply release-notes/release-notes53.md:65]: I guess this was very repetitive so it got refactored into the shorter section below? The headings appear to still exist though...

Comment (tylerkelly13) [reply release-notes/release-notes53.md:65]: Sorry, I was consolidating the note for these, and forgot to add something to these sections. Fixed in #1520

File: _config.yml (hunk 1 above)
Comment (tylerkelly13) [reply _config.yml:51]: > I guess we don't really care about hex as it's just an HCL bundle

HCL bundle has it's own docs elsewhere.

> Do you feel this wording is making it obvious that it's talking about each individual service war? e.g. ephox-spelling

only way to be 100% clear is to spell out each one explicitly. I'd rather not.

> Should we put it in a note of how to check your version?

Updated to:
```
Note: This feature is only available for TinyMCE self-hosted server-side components, version 2.3.0 and later. To check the version, visit <domain>/<service>/version. Such as http://localhost:8080/ephox-hyperlinking/version.
```

File: _data/nav.yml (hunk 2 above)
Comment (tylerkelly13) [reply _data/nav.yml:730]: ln 729: Noun
In 730: not a Noun

File: release-notes/release-notes53.md (hunk 5 above)
Comment (tylerkelly13) [reply release-notes/release-notes53.md:125]: We can, so I did.

File: enterprise/server/configure.md (hunk 3 above)
Comment (tylerkelly13) [reply enterprise/server/configure.md:136]: added a why: https://github.com/tinymce/tinymce-docs/pull/1518/commits/771c52fe3bd2e2c75e93dc057f9ccc6bf08a7dc0

File: _data/nav.yml (hunk 2 above)
Comment (tylerkelly13) [reply _data/nav.yml:730]: to expand, opting to use sentence case, rather than title case, so only nouns and first word get a leading capitalised letter

**Pull request merged**